isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -2746,6 +2828,206 @@ namespace isobus
+											}
+										}
+									}
+								}
+							}
+							else
+							{
+								bool found = false;
+								for (AuxiliaryInputDevice &aux : parentVT->auxiliaryInputDevices)
+								{
+									if (aux.name == isoName)
+									{
+										found = true;
+										if (static_cast<std::uint8_t>(AuxiliaryTypeTwoFunctionType::QuadratureBooleanMomentary) >= functionType)
+										{
+											AssignedAuxiliaryFunction assignment(functionObjectID, inputObjectID, static_cast<AuxiliaryTypeTwoFunctionType>(functionType));
+											auto location = std::find(aux.functions.begin(), aux.functions.end(), assignment);
+											if (aux.functions.end() == location)
+											{
+												aux.functions.push_back(assignment);
+												if (storeAsPreferred)
+												{
+													// TODO save preferred assignment to persistent configuration
+												}
+												CANStackLogger::CAN_stack_log("[AUX-N]: Assigned function " + isobus::to_string(static_cast<int>(functionObjectID)) + " to input " + isobus::to_string(static_cast<int>(inputObjectID)));
+											}
+											else
+											{
+												hasError = true;
+												isAlreadyAssigned = true;
+												CANStackLogger::CAN_stack_log("[AUX-N]: Unable to store preferred assignment due to missing auxiliary input device with name: " + isobus::to_string(isoName));
+											}
+										}
+										else
+										{
+											hasError = true;
+											CANStackLogger::CAN_stack_log("[AUX-N]: Unable to store preferred assignment due to unsupported function type: " + isobus::to_string(functionType));
+										}
+										break;
+									}
+								}
+								if (false == found)
+								{
+									hasError = true;
+									// TODO prettier logging of NAME
+									CANStackLogger::CAN_stack_log("[AUX-N]: Unable to store preferred assignment due to missing auxiliary input device with name: " + isobus::to_string(isoName));
+								}
+							}
+							parentVT->send_aux_n_assignment_response(functionObjectID, hasError, isAlreadyAssigned);
+						}
+						break;
+
+						case static_cast<std::uint8_t>(Function::AuxiliaryInputTypeTwoStatusMessage):
+						{
+							std::uint16_t inputObjectID = (static_cast<std::uint16_t>(data[1]) | ((static_cast<std::uint16_t>(data[2])) << 8));
+							std::uint16_t value1 = (static_cast<std::uint16_t>(data[3]) | ((static_cast<std::uint16_t>(data[4])) << 8));
+							std::uint16_t value2 = (static_cast<std::uint16_t>(data[5]) | ((static_cast<std::uint16_t>(data[6])) << 8));
+							bool learnModeActive = (0x01 == (data[7] & 0x01));
+							bool inputActive = (0x02 == (data[7] & 0x02)); // Only in learn mode?
+							bool controlIsLocked = false;
+							bool interactionWhileLocked = false;
+							if (parentVT->get_vt_version_supported(VTVersion::Version6))
+							{
+								controlIsLocked = (0x04 == (data[7] & 0x04));
+								interactionWhileLocked = (0x08 == (data[7] & 0x08));
```

Comment:
Automated finding by cppcheck:
Variable 'interactionWhileLocked' is assigned a value that is never used.

Reply:
Added:
```doxygen
/// @todo figure out how to best pass other status properties below to application
```